athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/common/PartitionUtil.java

```diff
@@ -0,0 +1,340 @@
+            case "smallint":
+            case "tinyint":
+                return "(\\d+)";
+            case "date":
+                String datePattern = tableParameters.get(String.format("partition.%s.pattern", columnName));
+                if (datePattern == null) {
+                    return "(" + DEFAULT_DATE_REGEX_STRING + ")";
+                }
+                return "(" + getDateRegExByPattern(datePattern) + ")";
+            default:
+                throw new IllegalArgumentException("Column type '" + columnType + "' is not supported for a partition column in this connector");
+        }
+    }
+
+    /**
+     * Return a column value with exact type from string value
+     *
+     * @param columnName  Name of the column
+     * @param columnType column type
+     * @param tableParameters   table parameters
+     * @return object of column value
+     */
+    private static Object convertStringByColumnType(String columnName, String columnType, String columnValue, Map<String, String> tableParameters) throws ParseException
+    {
+        switch (columnType) {
+            case "string":
+            case "varchar":
+                return columnValue;
+            case "bigint":
+                return Long.parseLong(columnValue);
+            case "int":
+            case "smallint":
+            case "tinyint":
+                return Integer.parseInt(columnValue);
+            case "date":
+                String datePattern = tableParameters.get(String.format("partition.%s.pattern", columnName));
+                if (datePattern == null) {
+                    datePattern = DEFAULT_DATE_REGEX_STRING;
+                }
+                return new SimpleDateFormat(datePattern).parse(columnValue);
+            default:
+                throw new IllegalArgumentException("Column type '" + columnType + "' is not supported for a partition column in this connector");
+        }
+    }
+
+    /**
+     * Return a date pattern string
+     *
+     * @param datePattern  value of partition folder when date pattern
+     * @return string of date pattern
+     */
+    private static String getDateRegExByPattern(String datePattern)
+    {
+        if (datePattern == null || datePattern.isBlank()) {
+            return datePattern;
+        }
+        return datePattern.replaceAll("[YyMDdFEHhkKmswWS]", "\\\\d")
+                .replaceAll("'", "") // replace ' from 'T'
+                .replaceAll("Z", "-\\\\d{3,4}") // replace time-zone offset with 3-4 digits with the prefix '-'
+                .replaceAll("z", "(.*?){2,6}") // replace time zone abbreviations with any character of length of min 2, max 6 (currently max is 5)
+                .replaceAll("G", "AD"); // Era designator. Currently BC not supported
+    }
+
+    /**
+     * Determine the partitions based on Glue Catalog
+     * @return A list of partitions
+     */
+    public static PartitionResult getPartitions(Table table, Map<String, FieldReader> fieldReadersMap)
```

Comment:
Can you name this function better as well? It doesn't seem like this function is performing what I would expect a function named `getPartitions` to do. This instead looks like its substituting in values into the table location + partition pattern?